core/process.py

```diff
@@ -399,10 +408,21 @@ def _ApplyRedirect(self, r):
                 try:
                     open_fd = posix.open(arg.filename, mode, 0o666)
                 except (IOError, OSError) as e:
-                    self.errfmt.Print_("Can't open %r: %s" %
-                                       (arg.filename, pyutil.strerror(e)),
-                                       blame_loc=r.op_loc)
-                    raise  # redirect failed
+                    if noclobber != 0 and e.errno == EEXIST:
+                        self.errfmt.PrintMessageEx(
+                            "I/O redirect error: can't overwrite existing file %r: %s"
+                            % (arg.filename, pyutil.strerror(e)),
+                            self.mutable_opts.Get(option_i.verbose_errexit),
+                            r.op_loc)
+                    else:
+                        self.errfmt.PrintMessageEx(
+                            "I/O redirect error: can't open file %r: %s" %
+                            (arg.filename, pyutil.strerror(e)),
+                            self.mutable_opts.Get(option_i.verbose_errexit),
+                            r.op_loc)
+                    raise IOError(
```

Comment:
Hm I don't think this is the right way to do it, we can maybe tackle this separately from noclobber

I agree the messages can be tweaked, but we should do it in a more systematic way


I use `test/runtime-errors.sh` to eyeball the errors

Reply:
the thing is there is error code clobber issue, this is why printing message and raisin a false error seemed a good idea.
otherwise, we'll need something other than IOError/OSError, but I wasn't able to achieve this as there are impacts with c++ I believe.
and as this is the only place in code that did double print message, I found this dirty trick.
but we want to keep caller to catch oserror coming from posix.close, write and so on.

Reply:
This IOError(0) trick is too subtle

As mentioned, we need to have some demos in `test/runtime-errors.sh` to tickle all the error cases that are affected

And I'd like to see examples of before and after

But let's do that in another change